packages/editor/CodeMirror/markdown/markdownReformatter.ts

```diff
@@ -644,15 +646,38 @@ export const renumberList = (state: EditorState, sel: SelectionRange): Selection
 
 			const filteredText = stripBlockquote(line);
 			const match = filteredText.match(listItemRegex);
+
+			// Skip lines that aren't the correct type (e.g. blank lines)
+			if (!match) {
+				continue;
+			}
+
 			const indentation = match[1];
 
 			const indentationLen = tabsToSpaces(state, indentation).length;
-			const targetIndentLen = tabsToSpaces(state, currentGroupIndentation).length;
+			let targetIndentLen = tabsToSpaces(state, currentGroupIndentation).length;
 			if (targetIndentLen < indentationLen) {
-				listNumberStack.push(nextListNumber);
+				listNumberStack.push({ nextListNumber, indentationLength: indentationLen });
 				nextListNumber = 1;
 			} else if (targetIndentLen > indentationLen) {
-				nextListNumber = listNumberStack.pop() ?? parseInt(match[2], 10);
+				nextListNumber = parseInt(match[2], 10);
+
+				// Handle the case where we deindent multiple times. For example,
+				// 1. test
+				//    1. test
+				//      1. test
+				// 2. test
+				while (targetIndentLen > indentationLen) {
+					const listNumberRecord = listNumberStack.pop();
+
+					if (!listNumberRecord) {
+						break;
+					} else {
+						targetIndentLen = listNumberRecord.indentationLength;
+						nextListNumber = listNumberRecord.nextListNumber;
+					}
+				}
```

Comment:
I'm unsure why this wasn't necessary to handle double de-indentation before.

I suspect previously sublists weren't renumbered with parent lists. However, selecting a parent list and pressing <kbd>tab</kbd> *should* have also caused sublists to be renumbered.